From aba104c76b2bf44dbf96a5db0948c2d8024496e7 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:21:15 -0400
Subject: [iommu] vt-d: Use for_each_drhd_unit() instead of list_for_each_entry()

Message-id: <20140810182115.24055.50365.stgit@gir.stowe>
Patchwork-id: 87589
O-Subject: [RHEL7.1 PATCH 18/19] iommu/vt-d: Use for_each_drhd_unit() instead of list_for_each_entry()
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 8b161f0ee911369232b5b5c18f84b5072f5bd09e

commit 8b161f0ee911369232b5b5c18f84b5072f5bd09e
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Oct 31 17:25:16 2013 +0800

    iommu/vt-d: Use for_each_drhd_unit() instead of list_for_each_entry()

    Use for_each_drhd_unit() instead of list_for_each_entry for
    better readability.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/dmar.c b/drivers/iommu/dmar.c
index 4ee523e..41a526b 100644
--- a/drivers/iommu/dmar.c
+++ b/drivers/iommu/dmar.c
@@ -403,7 +403,7 @@ dmar_find_matched_drhd_unit(struct pci_dev *dev)
 
  dev = pci_physfn(dev);
 
- list_for_each_entry(dmaru, &dmar_drhd_units, list) {
+ for_each_drhd_unit(dmaru) {
   drhd = container_of(dmaru->hdr,
         struct acpi_dmar_hardware_unit,
         header);
-- 
1.7.1